app/models/working_hour.rb

```diff
@@ -69,4 +71,14 @@ def close_after_open
 
     errors.add(:close_hour, 'Closing time cannot be before opening time')
   end
+
```

Comment:
let's also add. a validation to ensure both the open_all_day and closed_all_day parameters can be saved as true at the same time .

example:

```
 validate :open_all_day_and_closed_all_day

 def open_all_day_and_closed_all_day
    return unless open_all_day? && closed_all_day?
    errors.add(:base, "open_all_day and closed_all_day cannot be true at the same time")
 end
```

Reply:
Just a thought, the `ensure_open_all_day_hours` validation always updates `closed_all_day` to `false` if `open_all_day` is `true` right? So do we need this check? Or am I missing any scenarios?

Reply:
okay